### ISSUE TITLE
Editable fields + checkbox option for table component's rows

### ISSUE THREAD

**Issue by ToniJ123** [sqlpage/SQLPage#722]
**What are you building with SQLPage ?**

Large system

**What is your problem ? A description of the problem, not the solution you are proposing.

What are you currently doing ? Since your solution is not implemented in SQLPage currently, what are you doing instead ?**

Table component results as editable more multiple ways than only using markdown/icon to make a link to do something.
There could be option for text fields to be editable, and dropdown menu option to fields and options would be from database.
Date field easily editable straight to the row? Checkbox option for multiselect table rows (to do anything which each want's to do)

Just one kind of solution, picture will say more than thousand words, even this isn't the exactly which I mean, but you get the point.

![editable-table-component](https://github.com/user-attachments/assets/26ba7f4d-c7b8-4c4a-90ee-a1bca0fa9fbd)

**A clear and concise description of what you want to happen.**

Table component rows to be able to choose if they are editable, and which way (text field, dropdown, date, etc..), also input restrictions as option. Also checkbox option to multiselect rows for some external button's triggered event or some kind of mass editing.

Thoughts?

**Comment by lovasoa** [comment]
That's a big feature, and would indeed be very nice ! But that's currently not a priority, since most of this (and more) can be achieved with the existing [spreadsheet component](https://github.com/sqlpage/sqlpage-spreadsheet/).

**Comment by ToniJ123** [comment]
Yeah I know it's a big overhaul to table component if these features would be done there. Spreadsheet component is awesome, but you cannot have dropdown menu's for cells or anything like that at the moment, and this table component's list -type of view, would be better for some use cases. It's lot asked I know. Anybody else thoughs about this?
